FirebaseAI/Sources/Types/Public/Live/LiveSession.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// A live WebSocket session, capable of streaming content to and from the model.
+///
+/// Messages are streamed through ``responses``, and can be sent through either the dedicated
+/// realtime API function (such as ``sendAudioRealtime(audio:)`` or ``sendTextRealtime(text:)``), or
+/// through the incremental API (such as ``sendContent(_:turnComplete:)``).
+///
+/// To create an instance of this class, see ``LiveGenerativeModel``.
+@available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, watchOS 8.0, *)
+@available(watchOS, unavailable)
+public final class LiveSession: Sendable {
+  private let service: LiveSessionService
+
+  /// An asyncronous stream of messages from the server.
+  ///
+  /// These messages from the incremental updates from the model, for the current conversation.
+  public var responses: AsyncThrowingStream<LiveServerMessage, Error> { service.responses }
+
+  init(service: LiveSessionService) {
+    self.service = service
+  }
+
+  /// Response to a ``LiveServerToolCall`` received from the server.
+  ///
+  /// - Parameters:
+  ///   - responses: Client generated function results, matched to their respective
+  ///     ``FunctionCallPart`` by the `functionId` field.
+  public func functionResponses(_ responses: [FunctionResponsePart]) async {
+    let message = BidiGenerateContentToolResponse(
+      functionResponses: responses.map { $0.functionResponse }
+    )
+    await service.send(.toolResponse(message))
+  }
+
+  /// Sends an audio input stream to the model, using the realtime API.
+  ///
+  /// To learn more about audio formats, and the required state they should be provided in, see the
+  /// docs on
+  /// [Supported audio formats](https://cloud.google.com/vertex-ai/generative-ai/docs/live-api#supported-audio-formats).
+  ///
+  /// - Parameters:
+  ///   - audio: Raw 16-bit PCM audio at 16Hz, used to update the model on the client's
+  ///     conversation.
+  public func sendAudioRealtime(_ audio: Data) async {
+    // TODO: (b/443984790) address when we add RealtimeInputConfig support
+    let message = BidiGenerateContentRealtimeInput(
+      audio: InlineData(data: audio, mimeType: "audio/pcm")
+    )
+    await service.send(.realtimeInput(message))
+  }
+
+  /// Sends a video input stream to the model, using the realtime API.
+  ///
+  /// - Parameters:
+  ///   - video: Encoded video data, used to update the model on the client's conversation.
+  ///   - format: The format that the video was encoded in (eg; `mp4`, `webm`, `wmv`, etc.,).
+  public func sendVideoRealtime(_ video: Data, format: String) async {
```

Comment:
Have the other platforms implemented sending video already? Have we tested manually that it works?

Reply:
```suggestion
  func sendVideoRealtime(_ video: Data, format: String) async {
```

Reply:
Going to punt for more testing.

Reply:
Resolving.